src/workarounds/aarch64_mpi.jl

```diff
@@ -0,0 +1,90 @@
+"""
+MPI reduction opertations with custom types (i.e. anything that has not a MPI datatype equivalent)
+are not available on aarch64. These are temprorary workarounds, where variables with custom types 
+are broken down to standard types before communication, and recast to the initial types after.
+This file was created by fixing all MPI errors encountered by running the tests on an ARM machine:
+all sensible MPI reduction routines are implemented for each custom type causing an error.
+"""
+
+# Julia's Bool type has no direct equivalent MPI datatype => need integer conversion
+function mpi_min(bool::Bool, comm::MPI.Comm)
+    int = Int(bool)
+    Bool(mpi_min(int, comm))
+end
+
+function mpi_max(bool::Bool, comm::MPI.Comm)
+    int = Int(bool)
+    Bool(mpi_max(int, comm))
+end
+
+# Vec3{T} must be cast to Vector{T} before MPI reduction
+function mpi_sum!(arr::Vector{Vec3{T}}, comm::MPI.Comm) where{T}
+    n = length(arr)
```

Comment:
Can this not be solved by a reinterpret(reshape, ... )? Then there is no copy

Reply:
I think this one is tricky to get without copies, because `SVector` are immutable. So even if I use something like `new_aarr = reshape(reinterpret(T, arr), 3, :)`, I still could not call `mpi_sum!()`. And assuming I call `mpi_sum()` instead, I also end up with a copy.